backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +75,127 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author_with_config(author_data, matcher_config):
+    matched_records = [
+        matched_record
+        for matched_record in match(author_data, matcher_config)
+        if matched_record.get("inner_hits")
+    ]
```

Comment:
I did a list comprehension here and not generator because I reuse `matched_records` later with validators.